python/lsst/pipe/tasks/zooniverseCutouts.py

```diff
@@ -0,0 +1,224 @@
+            image = self.generate_image(science, template, difference, center)
+            filename = os.path.join(image_path, f"{int(source['diaSourceId'])}.png")
+            with open(filename, "wb") as outfile:
+                outfile.write(image.getbuffer())
+
+    def generate_image(self, science, template, difference, center):
+        """Get a 3-part cutout image to save to disk, for a single source.
+
+        Parameters
+        ----------
+        science : `lsst.afw.image.ExposureF`
+            Science exposure to include in the cutout.
+        template : `lsst.afw.image.ExposureF`
+            Matched template exposure to include in the cutout.
+        difference : `lsst.afw.image.ExposureF`
+             Matched science - template exposure to include in the cutout.
+        center : `lsst.geom.SpherePoint`
+            Center of the source to be cut out of each image.
+
+        Returns
+        -------
+        image: `io.BytesIO`
+            The generated image, to be output to a file or displayed on screen.
+        """
+        size = lsst.geom.Extent2I(self.config.size, self.config.size)
+        return self._plot_cutout(science.getCutout(center, size),
+                                 template.getCutout(center, size),
+                                 difference.getCutout(center, size)
+                                 )
+
+    def _plot_cutout(self, science, template, difference):
+        """Plot the cutouts for a source in one image.
+
+        Parameters
+        ----------
+        science : `lsst.afw.image.ExposureF`
+            Cutout Science exposure to include in the image.
+        template : `lsst.afw.image.ExposureF`
+            Cutout template exposure to include in the image.
+        difference : `lsst.afw.image.ExposureF`
+             Cutout science - template exposure to include in the image.
+
+        Returns
+        -------
+        image: `io.BytesIO`
+            The generated image, to be output to a file via
+            `image.write(filename)` or displayed on screen.
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib import cm
+        import astropy.visualization as aviz
+
+        # TODO: how do we color masked pixels (including edges)?
+
+        def do_one(ax, data, name):
+            """Plot a normalized image on an axis."""
+            if name == 'Difference':
+                norm = aviz.ImageNormalize(data, interval=aviz.ZScaleInterval(),
+                                           stretch=aviz.LinearStretch())
+            else:
+                norm = aviz.ImageNormalize(data, interval=aviz.MinMaxInterval(),
+                                           stretch=aviz.AsinhStretch(a=self.config.asinhScale))
+            ax.imshow(data, cmap=cm.bone, interpolation="none", norm=norm)
+            ax.axis('off')
+            ax.set_title(name)
+
+        f, (ax1, ax2, ax3) = plt.subplots(1, 3)
```

Comment:
I've renamed `f` to `fig`. subplots returns the figure and the axes, and `fig`, and `axN` are the matplotlib convention for those.

Reply:
Yes, but you don't seem to use `f`/`fig` anywhere -- it would be better if that were explicit instead of the reader wondering where they overlooked the reference.